### ISSUE TITLE
Model training fails with bug

### ISSUE THREAD

**Issue by suit4** [acconeer/acconeer-python-exploration#94]
Hi there,

I am trying to get the Machine Learning mode of the Exploration Tool to train models.

So far, I have configured a feature, created some frames (about 18), inspected these frames and remove bad ones.
Now I would like to train that model.

I load the session data with all the good frames from a file ml_session_data_2021_10_29_1444_swipe_right.npy
Feature inspection shows all good frames.
All frames have the same label.
Model parameters has seven layers.

I now load training data and use the same file.
A dialog pops up, saying:
```
Loaded training data with shape (17, 66, 48, 1)
Found labels:
swipe_right
```

Then I click on "Train".
This gives an error messages saying:
```
Failed to train model!

ml_gui_elements.py
4114
compute_class_weight() takes 1 positional argument but 3 were given
```

After some googling and debugging (https://discuss.pytorch.org/t/problems-iterating-through-loader-when-using-sampler/112157)
I changed in file keras_processing_tf2.py line 190 to:
`class_weights = class_weight.compute_class_weight("balanced", classes=np.unique(y_ints), y=y_ints)`

This seems to work better, but then again fails with
```
Failed to train model!

ml_gui_elements.py
4114
indices[0] = 1 is not in [0, 1)
[[{{node GatherV2}}]]
[[MultiDeviceIteratorGetNextFromShard]]
[[RemoteCall]]
[[IteratorGetNextAsOptional]] [Op:__inference_train_function_1336]

Function call stack:
train_function
```

Any ideas whats going wrong here?

**Environment (please complete the following information):**
 - Host OS: macOS Catalina
 - Acconeer hardware: Raspberry Pi XC112, connected via LAN
 - Acconeer software version: 2.8.2
 - Exploration version: 2.8.0
 - Python 3.8

**Comment by Pflo-acconeer** [comment]
Thank you for your investigation as well as your thourough description, it was very easy to replicate the issue!

I was able to replicate your error when using a single label for the training data and the error went away when using multiple labels. 
The error message that you received (indices[0] = 1 is not in [0, 1)) is not descriptive of the problem and the fix that you suggest remedies that.

**Comment by Pflo-acconeer** [comment]
Thank you for your investigation as well as your thorough description, it was very easy to replicate the issue!

I was able to replicate your error when using a single label for the training data and the error went away when using multiple labels. 
The error message that you received (indices[0] = 1 is not in [0, 1)) is not descriptive of the problem and the fix that you suggest remedies that.

**Comment by benjaminschn** [comment]
Hi, I have the same Error when clicking "Train Modell":

```
Failed to train model!

ml_gui_elements.py
4114
compute_class_weight() takes 1 positional argument but 3 were given
```

For me the fix from Suit4 sadly does not work since I already have 3 different labels.

![image](https://user-images.githubusercontent.com/93918126/140738399-0ea6f664-503a-485b-8065-d9914e3f8622.png)

Host OS: Windows 10
Acconeer hardware: Raspberry Pi + A111 via Ethernet
Acconeer software version: 2.8.2
Exploration version: 2.8.0
Python 3.8

**Comment by Pflo-acconeer** [comment]
Hello benjaminschn!

Did you apply this fix? :
> I changed in file keras_processing_tf2.py line 190 to:
> class_weights = class_weight.compute_class_weight("balanced", classes=np.unique(y_ints), y=y_ints)

Or the check that addresses the number of classes?

**Comment by benjaminschn** [comment]
Thank you for your answer.

I applied this fix:

```
I changed in file keras_processing_tf2.py line 190 to:
class_weights = class_weight.compute_class_weight("balanced", classes=np.unique(y_ints), y=y_ints)
```
and still got the error from my Screenshot.

**Comment by Pflo-acconeer** [comment]
The reason that I was a bit confused is that the error message:

> compute_class_weight() takes 1 positional argument but 3 were given

Should be gone when applying the aforementioned fix.

The screenshot that you provided doesn't indicate an error- could you provide a screenshot with the error message?

**Comment by benjaminschn** [comment]
Oh sorry, i mixed that up. Here is the Error:

![Screenshot 2021-11-09 165751](https://user-images.githubusercontent.com/93918126/140959979-03f496d4-2d67-4546-b46b-eacef601161a.jpg)

I don´t know if this is relevant, but i saved the data as 3 different sessions with each session containing only 1 type of label. I then loaded in the 3 Sessions at once.

**Comment by Pflo-acconeer** [comment]
The error message that you receive indicates that compute_class_weights() from sklearn is being sent three positional arguments. This was ok in earlier versions but removed later, so only the first argument can be positional (the later has to be keyworded).

So this could be due to your tensorflow version being low - you then import keras_processing_tf1.py, can you verify that the fix is executed so that you don't execute code from keras_processing_tf1.py

**Comment by benjaminschn** [comment]
I tried tensorflow version 2.6.0 and 2.7.0. 
How do i make sure that the fix is executed? I tried deleting keras_processing_tf1.py and got the same error.
